<launch>
  <include file="$(find autorally_core)/launch/hardware.machine" />

  <node pkg="nodelet" type="nodelet" name="AutoRallyChassis" args="load autorally_core/AutoRallyChassis autorally_core_manager" machine="autorally-master" output="screen">

    <param name="commandRate" value="75" />
    <param name="commandMaxAge" value="0.2" />
    <param name="runstopMaxAge" value="2" />
    <param name="wheelDiameter" value="0.190" />

    <rosparam param="actuators" command="load" file="$(env AR_CONFIG_PATH)/arChassisConfig_$(env AR_CHASSIS).yaml" />

    <rosparam param="chassisCommandProirities" command="load" file="$(env AR_CONFIG_PATH)/chassisCommandPriorities.yaml" />
    
    <!--configure settings for 115200 baud, 8N1 -->
    <param name="port" value="/dev/arChassis" />
    <param name="serialBaud" value="115200" />
    <param name="serialDataBits" value="8" />
    <param name="serialParity" value="none" />
    <param name="serialStopBits" value="1" />
    <param name="serialHardwareFlow" value="false" />
    <param name="serialSoftwareFlow" value="false" />
  </node>

</launch>
